File: backend/src/main/java/com/lms/backend/services/UserService.java
package com.lms.backend.services;

import java.util.List;

import org.springframework.core.io.Resource;
import org.springframework.web.multipart.MultipartFile;

import com.lms.backend.model.User;

public interface UserService {
    User createUser(User user);

    User createAdmin(User user);

    User updateUser(User user, String userId);

    User deleteUser(String userId);

    List<User> getUsers();

    List<User> getAdmins();

    User getUserById(String userId);

    User getUserByReferenceNumber(String referenceNumber);

    User approveUser(String userId);

    List<User> findAllPendingUser();

    String saveUserImage(String bookId, MultipartFile file);

    Resource getImage(String bookId);
}

File: frontend/src/components/services/userservice.ts
import User from "@/model/User";
import axios from "./api";

const BASE_URL = "/user";

export function createUser(user: User) {
  return axios.post(`${BASE_URL}/create`, user);
}

export function createAdmin(user: User) {
  return axios.post(`${BASE_URL}/createAdmin`, user);
}

export function updateUser(user: User, userId: string) {
  return axios.put(`${BASE_URL}/update/${userId}`, user);
}

export function deleteUser(userId: string) {
  return axios.delete(`${BASE_URL}/delete/${userId}`);
}

export function getUsers() {
  return axios.get(`${BASE_URL}/users`);
}

export function getAdmins() {
  return axios.get(`${BASE_URL}/admins`);
}

export function getUserById(userId: string) {
  return axios.get(`${BASE_URL}/${userId}`);
}

export function getUserByReferenceNumber(userReferenceNumber: string) {
  return axios.get(`${BASE_URL}/byreferencenumber/${userReferenceNumber}`);
}

export function approveUser(userId: string) {
  return axios.put(`${BASE_URL}/approve/${userId}`);
}

export function getAllPendingUser() {
  return axios.get(`${BASE_URL}/pending`);
}

export function uploadUserImage(userId: string, formData: FormData) {
  return axios.post(`${BASE_URL}/${userId}/upload/image`, formData, {
    headers: {
      "Content-Type": "multipart/form-data",
    },
  });
}
